include/yaml-cpp/traits.h

```diff
@@ -104,16 +106,30 @@ struct disable_if : public disable_if_c<Cond::value, T> {};
 }
 
 template <typename S, typename T>
-class is_streamable {
+struct is_streamable {
   template <typename SS, typename TT>
   static auto test(int)
       -> decltype(std::declval<SS&>() << std::declval<TT>(), std::true_type());
 
   template <typename, typename>
   static auto test(...) -> std::false_type;
 
- public:
   static const bool value = decltype(test<S, T>(0))::value;
 };
 
+template<typename Key, bool Streamable>
+struct streamable_to_string {
+  static std::string impl(const Key& key) {
+    std::stringstream ss;
+    ss << key;
+    return ss.str();
+  }
+};
+
+template<typename Key>
+struct streamable_to_string<Key, false> {
+  static std::string impl(const Key&) {
+    return std::string();
```

Comment:
A thought: maybe write "<unprintable>" here rather than an empty string?

Reply:
Yeah, that looks better.